okpt/io/config/validator.py

```diff
@@ -0,0 +1,103 @@
+"""Provides validation for various config files.
+
+Functions:
+    validate(): Validate the tool configuration.
+
+Exceptions:
+    ConfigurationError: An error in the tool configuration.
+
+"""
+
+import os
+from io import TextIOWrapper
+from typing import Any, Dict
+
+import cerberus
+
+from okpt.io import reader
+from okpt.io.config import parser
+
+_CURR_FILE_DIR = os.path.dirname(os.path.abspath(__file__))
```

Comment:
Do these need to be global variables? Or can this be avoided and placed in a constructor?

Reply:
Yeah they don't. I've moved them to the constructor instead.